x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupResponseTranslator.java

```diff
@@ -238,11 +238,41 @@ private static SearchResponse doCombineResponse(SearchResponse liveResponse, Lis
                 ? (InternalAggregations)liveResponse.getAggregations()
                 : InternalAggregations.EMPTY;
 
-        rolledResponses.forEach(r -> {
-            if (r == null || r.getAggregations() == null || r.getAggregations().asList().size() == 0) {
-                throw new RuntimeException("Expected to find aggregations in rollup response, but none found.");
+        int missingRollupAggs = rolledResponses.stream().mapToInt(searchResponse -> {
+            if (searchResponse == null
+                || searchResponse.getAggregations() == null
+                || searchResponse.getAggregations().asList().size() == 0) {
+                return 1;
             }
-        });
+            return 0;
+        }).sum();
+
+        // We had no rollup aggs, so there is nothing to process
+        if (missingRollupAggs == rolledResponses.size()) {
+            // If we had a live response (even if it was empty) just return that
+            if (liveResponse != null) {
+                return liveResponse;
+            } else {
+                // Otherwise we just had rollup, so build an empty response
+                InternalSearchResponse combinedInternal = new InternalSearchResponse(SearchHits.empty(),
+                    InternalAggregations.EMPTY, null, null,
+                    rolledResponses.stream().anyMatch(SearchResponse::isTimedOut),
+                    rolledResponses.stream().anyMatch(SearchResponse::isTimedOut),
+                    rolledResponses.stream().mapToInt(SearchResponse::getNumReducePhases).sum());
+
+                int totalShards = rolledResponses.stream().mapToInt(SearchResponse::getTotalShards).sum();
+                int sucessfulShards = rolledResponses.stream().mapToInt(SearchResponse::getSuccessfulShards).sum();
+                int skippedShards = rolledResponses.stream().mapToInt(SearchResponse::getSkippedShards).sum();
+                long took = rolledResponses.stream().mapToLong(r -> r.getTook().getMillis()).sum() ;
+
+                // Shard failures are ignored atm, so returning an empty array is fine
```

Comment:
we should preserve the shard failures in the response, let's do that in a follow up ? We should also sum the number of shards (rollup + normal) in the response.

Reply:
Yeah, right now we just [fail the search request](https://github.com/elastic/elasticsearch/blob/81c221d478fe4658d8722f4208dac7c17744f7b7/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupResponseTranslator.java#L206-L222) if there are _any_ failures, which is why we don't bother to send any back right now.  

But I agree it'd be better to send back the failures.  The above was done for simplicity at the time :)

>  We should also sum the number of shards (rollup + normal) in the response.

Hm actually, I should tweak this whole section.  I didn't sum up the shards because we return the live response if it exists.  But it should be a combined sum regardless of if live/rollup exist or not.  I'll tweak it a little to make this section better.